### PR TITLE
Add TPX1 detector uncertainty calculation module

### PR DESCRIPTION
**Pull request by KedoKudo**
## Summary

This PR adds uncertainty calculation support for TPX1 MCP detector-corrected neutron imaging data. The module computes variance estimates using the detector efficiency correction formula, allowing users to obtain counts with associated uncertainties for any region of interest. The implementation has been validated against Monte Carlo simulations and tested with real VENUS beamline data.

## References

- **EWM**: [14273](https://ornlrse.clm.ibmcloud.com/ccm/web/projects/Neutron%20Data%20Project%20(Change%20Management)#action=com.ibm.team.workitem.viewWorkItem&id=14273)

## Testing Instructions

1. **Review the validation notebook**: `notebooks/uncertainty_validation.ipynb` demonstrates the mathematical derivation and Monte Carlo validation of the uncertainty formula.

2. **Run the demo notebook on real data**: `notebooks/uncertainty_real_data_test.ipynb` shows the module in action with actual detector data.
   > **Note**: Reviewers will need to adjust the data path in the notebook to point to available TPX1 detector data.

---
## [Co-Pilot Summary]

This pull request introduces a new uncertainty calculation module for TPX1 MCP detector-corrected neutron imaging data, along with comprehensive unit tests to ensure correctness and robustness. The changes provide a well-documented, production-ready implementation for computing uncertainty estimates, including raw count recovery and ROI aggregation, and establish a strong foundation for future development and integration.

**Core functionality for detector uncertainty calculations:**

* Added the `ibeatles.core.detector.uncertainty` module, which implements:
  - `compute_counts_with_uncertainty`: Main function to compute counts and uncertainties for a region of interest, given corrected TIFF stacks and shutter metadata.
  - `load_shutter_counts`: Utility to parse and validate shutter counts files, filtering out invalid entries and calculating normalization ratios.
  - Internal methods for raw count recovery (`_recover_raw_counts`) and variance computation (`_compute_variance`), following validated physical models and including robust input validation and error handling.

* Created a detector-specific package initializer (`ibeatles.core.detector.__init__.py`) that exposes the main uncertainty calculation functions for external use.

**Testing and validation:**

* Added comprehensive unit tests for all major functions in `test_uncertainty.py`, covering:
  - File parsing, error handling, and edge cases for shutter counts.
  - Mathematical correctness and monotonicity for raw count and occupancy recovery.
  - Variance formula validation and non-negativity.
  - End-to-end tests for ROI handling, shape mismatches, and CSV export.
  - Integration tests for round-trip consistency and workflow simulation.

* Included a minimal test module initializer for the detector unit test package.

### CODE REVIEW

File: src/ibeatles/core/detector/uncertainty.py [hunk 1]
```diff
@@ -0,0 +1,339 @@
...
+        DataFrame with columns:
+        - tof: Time-of-flight values
+        - counts: Summed counts in ROI
+        - uncertainty: Uncertainty (standard deviation) of summed counts
+
+    Examples
+    --------
+    >>> # Load data using existing iBeatles functions
+    >>> data = load_data_from_folder("/path/to/corrected/tiffs")
+    >>> spectra = load_time_spectra(spectra_file, distance, offset)
```
[the remark this answers is not in the document]
Comment (KedoKudo) [reply src/ibeatles/core/detector/uncertainty.py:245]: the docstring is sufficient

**Review by JeanBilheux — APPROVED** [review]
